python/mxnet/gluon/block.py

```diff
@@ -1175,7 +1175,14 @@ def _call_cached_op(self, *args):
             out = [out]
         return _regroup(out, self._out_format)
 
-    def optimize_for(self, x, *args, backend=None, backend_opts=None, clear=True, partition_if_dynamic=False, **kwargs):
+    def optimize_for(self, x, *args, backend=None, clear=False,
```

Comment:
Building on @waytrue17 's comment, is there any point to calling optimize_for without a backend? or should backend always be required (and we shouldnt set default to be None). What do you think @mseth10?

Reply:
That's a good point, we should make it always required